pkg/util/conc/future.go

```diff
@@ -71,6 +71,11 @@ func (future *Future[T]) Err() error {
 	return future.err
 }
 
+func (future *Future[T]) SetErr(err error) {
```

Comment:
This method may be abused to set error of future multiple times, but the channel can't be closed twice